Source/Core/Core/DSP/DSPHWInterface.cpp

```diff
@@ -46,25 +46,25 @@ void gdsp_ifx_init()
 		g_dsp.ifx_regs[i] = 0;
 	}
 
-	g_dsp.mbox[0] = 0;
-	g_dsp.mbox[1] = 0;
+	g_dsp.mbox[0].store(0);
+	g_dsp.mbox[1].store(0);
 }
 
 u32 gdsp_mbox_peek(u8 mbx)
 {
-	return Common::AtomicLoad(g_dsp.mbox[mbx]);
+	return g_dsp.mbox[mbx].load();
 }
 
 void gdsp_mbox_write_h(u8 mbx, u16 val)
 {
-	const u32 new_value = (Common::AtomicLoadAcquire(g_dsp.mbox[mbx]) & 0xffff) | (val << 16);
-	Common::AtomicStoreRelease(g_dsp.mbox[mbx], new_value & ~0x80000000);
+	const u32 new_value = (g_dsp.mbox[mbx].load() & 0xffff) | (val << 16);
```

Comment:
Can you provide a small example of the correct way of doing it with compare_and_exchange (if you feel like doing so)? I've been reading up on multithreading and atomics recently to try and clean up instances where volatile is used in the codebase where threading-specific facilities are more apt, but I haven't researched/read that far into atomics (embarrassingly enough).